p2p/sync.go

```diff
@@ -293,8 +292,8 @@ func (s *syncService) adaptAndSanityCheckBlock(ctx context.Context, header *spec
 			for i, r := range receipts {
 				txHash := coreTxs[i].Hash()
 				if txHash == nil {
-					spew.Dump(coreTxs[i])
-					panic(fmt.Errorf("TX hash %d is nil", i))
+					s.log.Errorw("Transaction hash is nil", "index", i)
+					return
```

Comment:
Panic acted like an assertion here. The reason it was introduced in the first place that there were from time to time different problems related to final hash, but since then it became more stable so TL;DR: let's remove this check completely